### PR TITLE
[webgpu] Add support for SparseToDense

### PR DESCRIPTION
**Pull request by haoyunfeix**
FEATURE

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5629)

### CODE REVIEW

**Comment by haoyunfeix** [comment]
@qjia7 @axinging @xhcao @gyagp, PTAL

File: tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  constructor(
+      updateSize: number, sliceDim: number, indicesRank: number,
+      updatesRank: number, strides: number[], shape: number[],
+      summingDupeIndex = true) {
+    this.outputShape = shape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.shaderKey = `scatter_${indicesRank}_${updatesRank}_${strides}`;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts:51]: `strides` is already a uniform. Not needed to be part of the shaderKey?

File: tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  outputShape: number[];
+  userCode: string;
+  shaderKey: string;
+  dispatchLayout: {x: number[]};
+  dispatch: [number, number, number];
+  workGroupSize: [number, number, number] = [64, 1, 1];
+  size: number;
+  useWgsl: boolean;
+  sliceDim: number;
+  shape: number[];
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts:38]: nit: sliceDim/shape are never used members?

Comment (haoyunfeix) [reply tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts]: Done.

File: tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts (hunk 1 above)
Comment (haoyunfeix) [reply tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts]: Done.

File: tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import {computeDispatch, flatDispatchLayout} from '../webgpu_util';
+
+import {getUseWgsl, WebGPUProgram} from './webgpu_program';
+
+export class ScatterProgram implements WebGPUProgram {
+  variableNames = ['updates', 'indices', 'defaultValue'];
+  uniforms: string;
+  uniformsWgsl: string;
+  outputShape: number[];
+  userCode: string;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/scatter_webgpu.ts:30]: nit: `userCode` is not needed.

**Review by qjia7 — APPROVED** [review]
LGTM. Please resolve the conflicts.

**Review by pyu10055 — APPROVED** [review]
> Reviewed 3 of 4 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5629)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @haoyunfeix and @qjia7)

**Pull request merged**